app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,34 @@
+class PaymentsController < ApplicationController
+
+  def create
+    @product = Product.find(params[:product_id])
+    @user = current_user
+
+    token = params[:stripeToken]
+    begin
+      charge = Stripe::Charge.create(
+        amount: (@product.price.to_i * 100),
```

Comment:
`@product.price` is a string in your database. `#to_i` changes it into an integer so you can * 100.
Stripe API reads integers as pennies, thus:
100 * 100 = $100.00